src/functionBased/components/TodoItem.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable react/destructuring-assignment */
+import React, { useState, useEffect } from 'react';
+import { FaTrash } from 'react-icons/fa';
+import PropTypes from 'prop-types';
+import styles from './TodoItem.module.css';
+
+const TodoItem = (props) => {
+  const [editing, setEditing] = useState(false);
+
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  const handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      setEditing(false);
+    }
+  };
+
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+
+  // eslint-disable-next-line react/prop-types
+  const { completed, id, title } = props.todo;
+
+  const viewMode = {};
+  const editMode = {};
+
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  useEffect(
+    () => () => {
+      console.log('Cleaning up...');
+    },
+    [],
+  );
```

Comment:
I think here we are making extra unnecessary work, Could you please get rid of it? 
I know it was part of the tutorial for educational reasons but I think you forgot to remove it :)